qa/rpc-tests/blockchain.py

```diff
@@ -215,8 +219,39 @@ def _test_rollbackchain(self):
         self.nodes[0].rollbackchain(self.nodes[0].getblockcount() - 20)
 
         # Reconsider the fork2. Blocks should now be fully reconnected on fork2.
-        self.nodes[0].reconsiderblock(bestblockhash2)
-        assert_equal(self.nodes[0].getbestblockhash(), bestblockhash2);
+        self.nodes[0].reconsiderblock(bestblockhashfork2)
+        assert_equal(self.nodes[0].getbestblockhash(), bestblockhashfork2);
+
+
+        #### Start testing reconsidermostworkchain
+        # Create an additional fork 3 which is the longest fork. Then make the shortest
+        # fork2 the active chain.  Then do a reconsidermostworkchain which should make
+        # fork3 the active chain, and disregarding fork1 which is longer than fork2 but
+        # shorter than fork3.
+        
+        print ("Test that reconsidermostworkchain() works")
+
+        # rollback to before fork 1 and 2, and then mine another longer fork 3
+        self.nodes[0].rollbackchain(self.nodes[0].getblockcount() - 120, True)
+        fork3blocks = 140;
+        self.nodes[0].generate(fork3blocks)
+        bestblockhashfork3 = self.nodes[0].getbestblockhash() #save for later
+
+        # now rollback again and make the shortest fork2 the active chain
+        self.nodes[0].rollbackchain(self.nodes[0].getblockcount() - fork3blocks, True)
+        self.nodes[0].reconsiderblock(bestblockhashfork2)
+
+        # do a reconsidermostworkchain but without the override flag
+        try:
+            self.nodes[0].reconsidermostworkchain()
```

Comment:
fail the test if an exception is not thrown

Reply:
it does on line 249...the test here is just to check if the text is correct and that it wouldn't roll back the chain.

Reply:
But if reconsidermostworkchain() does not throw an exception then this check fails silently because your assert on line 250 is never run.

Reply:
oh yes you're right, duh...will fix